File: Competitve Programming/Templates to use/Binary  Search/binarySearch.cpp
#include <bits/stdc++.h>
#include <fstream>
using namespace std;

int arr[]={-4,-3,-2,-1,0,1,2};
bool bsCmp(int i){
    return arr[i] > -1;
}
//find the minimum i such that f(i) is true
bool findMin=true;
int binarySearch(int l, int r) { 
    int m;
    while (l <= r) { 
        m = (l+r)/ 2; 
        bool cmp = findMin?bsCmp(m):!bsCmp(m);
        if (cmp){r = m-1;}
        else{l = m + 1;}
    }
    return l; 
} 
int main(){
    //ios_base::sync_with_stdio(false);
    //cin.tie(NULL);
    
    cout<<"Running:";
    //cin>>arr[5];
    cout<<binarySearch(1,5);
    return 0;
}

File: School Exercises/3. H2 Computing Notes/Searching and Sorting/Binary_Search.py
array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
def binarySearch(array, low, high, find):
    while low <= high:
        mid = int((low+high)/2)
        if array[mid] == find:
            return mid
        elif array[mid] > find:
            high = mid - 1
        elif array[mid] < find:
            low = mid + 1
    return -1

print(binarySearch(array, 0, 9, 2.5))
    
